Adds a new netherworld critter, the imlagre

Armhulen opened the pull request:
The imlagre always comes in a group of three, and they are life linked to each other. When one dies, it will revive with half hp after 10 seconds if there is at least one other is left alive.

things I gotta do
add them retreating if they're the last one
~~hire a very qt spriter gf (i only have placeholders help)~~
~~add sounds~~

Commit:
welcome to the squad imlagre!

Commit:
WHAT A STEAL

Commit:
i'm gay and i'll fix this l8r

vuonojenmustaturska commented:
but are they large

DaedalusGame commented:
![golden_knight](https://user-images.githubusercontent.com/9559216/35930974-b3d8da60-0c33-11e8-9ff1-e5cc2cb76a0b.png)

Ratvar commented:
but are they lagre

XDTM commented:
Book of stabbing when

PKPenguin321 commented:
IM LARGE

Armhulen commented:
@XDTM DONT LET THEM KNOW

Armhulen commented:
the sprites are place holders until someone gives me a real one please don't cry

MMMiracles commented:
shouldn't it be pink, since the prior two netherworld mobs are also this pink/red color?

Armhulen commented:
nether is a reality humans can't even comprehend, so the creatures in it come in every colors. Keep in mind that the green creature found in the wizard den is a netherworld creature

Bawhoppen commented:
Cosmic horrors have no relation though

Armhulen commented:
can you say the maybe differently perhaps

Armhulen commented:
i don't get it

MMMiracles commented:
![lil guy](https://user-images.githubusercontent.com/9276171/35949916-e82e24d8-0c41-11e8-889a-05b652686d89.png)
look at him go, all tiny and blue

Bawhoppen commented:
The type creature Experiment 35b (in the wiz den) is known as a "cosmic horror"
They have no relation to the netherworld

ChangelingRain commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,66 @@
+		visible_message("<span class='danger'>[src]'s corpse suddenly explodes in a shower of gore!</span>")
+		i.visible_message("<span class='danger'>[i]'s corpse suddenly explodes in a shower of gore!</span>")
+		gib()
+		i.gib()
```
I think this might cause problems? not sure offhand.

Armhulen replied:
well what do you mean

ChangelingRain replied:
actually this proc has problems in general, if you kill two, and they're oriented in a certain way in the list, they'll both gib even if the third is alive

Armhulen replied:
i know, i've been trying to this fix and i have a pretty good idea on what i need to do

ExcessiveUseOfCobblestone replied:
His new code ((((should)))) handle this hence his addition of the args in death

he just needs to sync to the pr :^)

Armhulen commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,66 @@
+		two.linked_imlagre.Add(one)
+
+/mob/living/simple_animal/hostile/netherworld/imlagre/Life()
+	..()
```
i'm just a teeny bit curious on what the difference is between these

ChangelingRain replied:
. = ..() means that the lower life proc will properly return a value that very few things care about tbh

Armhulen replied:
thank you!

Commit:
fix!

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,65 @@
+		one.linked_imlagre.Add(src)
+		one.linked_imlagre.Add(two)
+		two.linked_imlagre.Add(src)
+		two.linked_imlagre.Add(one)
```
I showed him how to change this to also allow for a variety of linked imlagre amounts

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,65 @@
+	deathmessage = "unwinds in a a paroxysm of laughter."
+	var/list/linked_imlagre = list()
+	var/laughmod = 1 //need this on the mob so it carries over to life, and i like letting admins var as much as they please
+	var/list/laughs = list('sound/voice/human/manlaugh1.ogg', 'sound/voice/human/manlaugh2.ogg')
```
thanks lol

Armhulen commented:
@bawwwwwwwww


They do though! That is a netherworld mob, it's in the netherworld file and it's spawned by the netherworld portal

Bawhoppen commented:
It's been in the game since goon-era. You're the one to have moved it to the netherworld file, so unless it's always been some reference I didn't know about, that is a recent development and you're trying to retcon it into something else.

Dorsisdwarf commented:
I didnt know anything other than mi-gos got spawned by those portals.

ExcessiveUseOfCobblestone commented:
who cares?

Armhulen commented:
@bawwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww

yes, yes i am. the creature exactly captures what i wanted to do with the netherworld that i added and they had no reason to exist without this, and they had the same stats as mi-go so i just added it to where mi-gos are and added that they exist from the netherworld. cosmic horror and netherworld are really just synonyms except the creature isn't a random bloat mob, now it's the child of all nether mobs. much better!

Armhulen commented:
@bawwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww

that's how long i can hold my breath

Armhulen commented:
@baw that's how long i can stand all this talk about derelict mobs that sit in the wizard's den

Armhulen commented:
sorry brian weiser

Bawhoppen commented:
Sorry Arm, I just got so caught up on the technicalities, I forgot to see the big picture

ExcessiveUseOfCobblestone commented:
want to see a big picture :eggplant:

Armhulen commented:
my prs are shitpost friendly I paid shiz to look the other way

Armhulen commented:
I can't think of a good way to make sure there isn't one fucking off to a corner of the station without either ruining gameplay or ruining my will to code

Armhulen commented:
I can't think of a good way to make sure there isn't one fucking off to a corner of the station without either ruining gameplay or ruining my will to code

Armhulen commented:
throws some mf gang signs or some shit

lets revive this, i have a quick fix for the early gibbing issues but i need to contact cobby on the full fix

Armhulen commented:
i mean we could just stick with mine but coding that allows admins to also have fun is super important

and shitcode i guess

Commit:
i am fueled by dan gheesling

Commit:
they can now spawn from the portal

Commit:
Merge branch 'master' into patch-47

Commit:
fixes

Commit:
Merge branch 'patch-47' of https://github.com/Armhulen/tgstation into patch-47

ExcessiveUseOfCobblestone commented:
lol

ExcessiveUseOfCobblestone commented:
You'll want to find a *nice* way that makes them take damage / teleports them if they're out of sight from one another.

Armhulen commented:
My idea is to add two locators of the other imlagres to the hud, and they don't heal if they aren't in range of each other, maybe a chat between them but is it really needed

making one sentient offers the other two to ghosts to hopefully remedy some issues with playing with bots only counter strike source did it right

Commit:
fixes here, fixes there, fixes everywhere

Armhulen commented:
also, i finally have sprites for these, they now look like three eyed alien apes which works for me

Commit:
i forget what i changed here but i bet it was a fix

ExcessiveUseOfCobblestone commented:
https://github.com/Armhulen/tgstation/blame/6700c6e560193d5793fef54d4eb6f91f576ddfd2/code/modules/mob/living/simple_animal/hostile/netherworld.dm#L108

:/

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,68 @@
+/mob/living/simple_animal/hostile/netherworld/imlagre/proc/imlagre_revive()
+	var/itlives = FALSE
+	for(var/mob/living/simple_animal/hostile/netherworld/imlagre/i in linked_imlagre)
+		if(!i.stat) //if any of them are alive then REVIVE!!
```
Stats aren't boos don't do this.

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,68 @@
+			var/flufftext = list("wicked", "sinister", "baleful", "hideous", "wild", "malevolent")
+			visible_message("<span class='danger'>[i] winds back together with a [pick(flufftext)] cackle!</span>")
+			adjustBruteLoss(maxHealth * 0.5) //revived at half HP, or, about a carpsful of health
+	if(!itlives == TRUE)
```
haha

Armhulen replied:
hahahah fuck

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,68 @@
+
+/mob/living/simple_animal/hostile/netherworld/imlagre/Life()
+	. = ..()
+	if(!stat)
```
Not a boo

Armhulen replied:
I don't follow it worked for migo

ExcessiveUseOfCobblestone replied:
It's not wrong codewise (it works just fine as is), it's wrong formatting wise.

You can have stat be more than just 1 and 0.

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,68 @@
+	attacktext = "slashes"
+	deathmessage = "unwinds in a a paroxysm of laughter."
+	var/list/linked_imlagre = list()
+	var/laughmod = 1 //need this on the mob so it carries over to life, and i like letting admins var as much as they please
```
that and it's static post initialize.

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,68 @@
+			i.visible_message("<span class='danger'>[i]'s corpse explodes in a shower of gore!</span>")
+			i.gib()
+		visible_message("<span class='danger'>[src]'s corpse explodes in a shower of gore!</span>")
+		gib() //i don't think it matters that i do this last, but it saves me at least one compile finding out that gibbing before finishing messes the code up
```
whatever you're looping through should delete last so yes it matters.

Armhulen replied:
HA

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,68 @@
+
+/mob/living/simple_animal/hostile/netherworld/imlagre/proc/imlagre_revive()
+	var/itlives = FALSE
+	for(var/mob/living/simple_animal/hostile/netherworld/imlagre/i in linked_imlagre)
```
see review summary

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/netherworld.dm:
```diff
@@ -50,6 +50,68 @@
+			visible_message("<span class='danger'>[i] winds back together with a [pick(flufftext)] cackle!</span>")
+			adjustBruteLoss(maxHealth * 0.5) //revived at half HP, or, about a carpsful of health
+	if(!itlives == TRUE)
+		for(var/mob/living/simple_animal/hostile/netherworld/imlagre/i in linked_imlagre)
```
review summary

ExcessiveUseOfCobblestone reviewed (changes requested):
You want the for loops in the revive proc to be typeless as well (see Contributing.md)

ExcessiveUseOfCobblestone committed:
Improves Spawning

Commit:
Merge pull request #10 from ExcessiveUseOfCobblestone/patch-35

Improves Spawning

Commit:
AUGHGGSG

Commit:
Merge branch 'patch-47' of https://github.com/Armhulen/tgstation into patch-47

ExcessiveUseOfCobblestone committed:
tfw cant spell

Commit:
tfw cant code

Commit:
sigh ok.

Armhulen commented:
New in this commit

Sprites
Check link
Link talking
Broken dreams and spans

Next, just gotta fix the front facing Sprite and add some new sounds, beams, and Sprite the action button. Whew!

Commit:
now a range they need to be in to revive nice

Commit:
FIXES?

Armhulen commented:
i'm about to leave for a good long while with no computer so i can't work on this, it'll get stale

so for the second time...

81Denton commented:
:(
